### PR TITLE
[Serve] Add replica tag to request counter and error counter

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
After looking at the implementation, the metrics `serve_backend_request_counter` and `serve_backend_error_counter` seem to be scoped to per-replica, not per-backend, but we just weren't tagging them with the replica.  This PR adds the replica tag to these metrics.  It might make sense to rename these metrics with `replica` instead of `backend` but I think we are changing them to `deployment` soon anyway.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/backend_worker.py [hunk 1]
```diff
@@ -145,7 +145,10 @@ def __init__(self, _callable: Callable, backend_config: BackendConfig,
             description=("The number of queries that have been "
                          "processed in this replica."),
             tag_keys=("backend", ))
```
Comment (simon-mo) [inline python/ray/serve/backend_worker.py:147]: you need to specify the tag_keys here to include replica

File: python/ray/serve/backend_worker.py [hunk 2]
```diff
@@ -160,9 +163,12 @@ def __init__(self, _callable: Callable, backend_config: BackendConfig,
         self.error_counter = metrics.Counter(
             "serve_backend_error_counter",
             description=("The number of exceptions that have "
-                         "occurred in the backend."),
+                         "occurred in this replica."),
             tag_keys=("backend", ))
```
Comment (simon-mo) [inline python/ray/serve/backend_worker.py:167]: add to tag_keys

**Review by simon-mo — CHANGES_REQUESTED** [review]

**Comment by edoakes** [comment]
@architkulkarni test is failing because the replica isn't added in the tag_keys:
https://buildkite.com/ray-project/ray-builders-pr/builds/10728#74d803bf-bdba-4be4-9a9f-7ac0c4931730/6-8366

**Comment by edoakes** [comment]
Oh yeah what Simon said ^

**Comment by architkulkarni** [comment]
The failing test is //python/ray/tests:test_kill_raylet_signal_log on windows, which is known flaky on master

**Review by simon-mo — APPROVED** [review]

**Pull request merged**